README.md

```diff
@@ -1,2 +1,102 @@
-# scalajs-fs2-router
-Scalajs router for Single Page Webapplications based on fs2
+# fs2-spa-router
+
+Scalajs router for Single Page web-applications based on *fs2* using 
+*http4s-dsl*'s `Uri` and `Path` to calculate and represent routes.
+
+```scala
+"com.github.jberggg" %%% "fs2-spa-router" % "0.1.0-SNAPSHOT"
+```
+
+**Note**: CI/CD is not yet in place, you need to checkout the project and build 
+it yourself with `sbt publishLocal`.
+
+## How it works
+
+The router is built upon the [*History API*](https://developer.mozilla.org/en-US/docs/Web/API/History) 
+which allows to route on regular looking paths rather then a path section delimited
+by a `#`. 
+
+When navigating from within the SPA app, the path and state is pushed to the *History API*. 
+When the user presses the back button, the router will receive the previous path and state
+from the *History API* which then can be used  to reconstruct the view the user wants to navigate
+to. The library makes no assumption about the state pushed to and received from the *History Api*. 
+Thus it can be anything you want [as long as](https://developer.mozilla.org/en-US/docs/Web/API/History_API/Working_with_the_History_API):
+
+> The state object can be anything that can be serialized.
+
+## How to use
+
+You can use the `RouterService` to setup the routing like so:
+
+```scala
+import org.http4s._, org.http4s.dsl.io._, org.http4s.implicits._
+import cats.syntax.all._
+import com.github.jberggg.router.{RouterService, RouterDsl}
+import com.github.jberggg.router.syntax._
+/* ... plus some more imports */
+
+object MyApp {
+
+    trait MyAppState
+    object EmptyState extends MyAppState
+    case class CreateFormState(name: String, address: String) extends MyAppState
+    // and so on...
+
+    def run[F[_] : Monad : Async ]: F[ExitCode] =
+
+        RouterService.createSignalAndResources[F](EmptyState.toJsObject).use{ pathSignal =>    
+                    
+          // ... or us the Signal directly by consuming it with `discrete` 
+          // and add new path and state with `set`
+          implicit val routerDslInterpreter = RouterDsl(pathSignal)
```

Comment:
Out of curiosity why expose the underlying `Signal`? It's almost an implementation detail; perhaps it would be more user-friendly to directly create the `RouterDsl`.

Reply:
Because most likely not everyone wants to use the tagless final approach and because I was able to make the History API related logic independent of the DSL, the DSL is merely a wrapper for the Signal anyway and adds convenience and readability when working in a tagless-final style.